pkg/storage/ec/client.go

```diff
@@ -88,6 +88,11 @@ func (ec *ecClient) Put(ctx context.Context, nodes []*pb.Node, rs eestream.Redun
 	infos := make(chan info, len(nodes))
 
 	for i, n := range nodes {
+		if n == nil {
+			infos <- info{i: i, err: Error.New("Nil node")}
```

Comment:
Perhaps, don't set an error. This is not an error condition.

Reply:
Then we should rethink what a successful Put is - the if-check down at line 146. I think we should change the if-check to:
```go
if successfulCount + nilCount < rs.RepairThreshold()
```
where `nilCount` is the number of nil nodes.

Assuming that the nil nodes are the healthy nodes, we would have a successful repair Put if the number of successfully repaired nodes + healthy nodes reach or exceed the repair threshold.

Reply:
I added an error because of this code
`successfulNodes = make([]*pb.Node, len(nodes))
	var successfulCount int
	for range nodes {
		info := <-infos
		if info.err == nil {
			successfulNodes[info.i] = nodes[info.i]
			successfulCount++
		}
	}`

Reply:
I added an error because of this:
```successfulNodes = make([]*pb.Node, len(nodes))
	var successfulCount int
	for range nodes {
		info := <-infos
		if info.err == nil {
			successfulNodes[info.i] = nodes[info.i]
			successfulCount++
		}
	}```

Reply:
Oh, on a second thought, we don't actually need to modify that if-clause and introduce `nilCount`. If we don't return error to the channel (as requested in my initial comment) then nil nodes will be automatically counted in the `successfulNodes`.

Reply:
I think it's OK to add nil nodes to successfulCount. Nil nodes are the healthy nodes, so it should not corrupt the logic.

Reply:
Also, I think we need to read and discard the pieces generated by the EncodeReader for these nil nodes. Otherwise, we may hit some undesired behaviour in the EncodeReader.

I suggest to move this check inside the goroutine and do something like:

```go
if n == nil {
	_, err := io.Copy(ioutil.Discard, readers[i])
	infos <- info{i: i, err: err}
}
```